src/engraving/rendering/editMode/editmodedraw.cpp

```diff
@@ -0,0 +1,531 @@
+    case ElementType::INSTRUMENT_CHANGE:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::INSTRUMENT_NAME:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::JUMP:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::LAISSEZ_VIB_SEGMENT:
+        drawSlurTieSegment(item_cast<const SlurTieSegment*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::LYRICS:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::MMREST_RANGE:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::MARKER:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::MEASURE_NUMBER:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::PARTIAL_TIE_SEGMENT:
+        drawSlurTieSegment(item_cast<const SlurTieSegment*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::PLAYTECH_ANNOTATION:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::REHEARSAL_MARK:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::SLUR_SEGMENT:
+        drawSlurTieSegment(item_cast<const SlurTieSegment*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::STAFF_TEXT:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::STICKING:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::STRING_TUNINGS:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::SYSTEM_TEXT:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::TEMPO_TEXT:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::TEXT:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::TIE_SEGMENT:
+        drawSlurTieSegment(item_cast<const SlurTieSegment*>(item), painter, ed, currentViewScaling);
+        break;
+    case ElementType::TRIPLET_FEEL:
+        drawTextBase(item_cast<const TextBase*>(item), painter, ed, currentViewScaling);
+        break;
+    default:
+        drawEngravingItem(item, painter, ed, currentViewScaling);
```

Comment:
The other renderers require us to add an option for each `ElementType`, otherwise we hit an error. Should we do the same here?

Reply:
Do you mean we should add an assertion or "UNREACHABLE" here? I don't think that's necessary, because when this method is called with an element that does not have any special edit mode drawing, i.e. that is not handled explicitly here, that's not necessarily an error.